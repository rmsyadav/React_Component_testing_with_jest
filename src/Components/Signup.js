import { useNavigate } from "react-router-dom";
import { useFormik } from 'formik';
import { dateConvertIntoMiliSecond } from "../Utility/DateConversion";
import { useState } from "react";
import validate from "../Utility/validateRegistrationForm";
import axios from "axios";
import "../Stylesheet/Signup.css";
import Footer from "./CommonComponents/Footer";
const Signup =()=>{
    const navigate= useNavigate();
    const [isSuccessRegister,setIsSuccessRegister] = useState();
    const formik = useFormik({
        initialValues: {
          name: '',
          email: '',
          birthdayDate: '',
          gender:'',
          telphone:'',
          password:'',
          rePassword:''

        },
        validate,
        onSubmit: async(values) => {
            const userDeatils ={
                username:values.name,
                email:values.email,
                dob:dateConvertIntoMiliSecond(values.birthdayDate),
                gender:values.gender,
                phone:values.telphone,
                password:values.password
            }
          const response = await axios.post('/shopping/api/v1/register',userDeatils);
          if(response.data.statusCode === 200 && response.data.statusMessage === "success!" )
          {
            setIsSuccessRegister("success")
            navigate('/signin');
          }
          else{
            setIsSuccessRegister("error")
          }
        },
      });
    return(<>
         <section className="container-fluid bg-image rounded-3" style={{backgroundImage: "url('https://mdbcdn.b-cdn.net/img/Photos/new-templates/search-box/img4.webp')"}}>
            <div className="mask py-5 d-flex align-items-center container gradient-custom-3">
                <div className="container">
                <div className="row d-flex justify-content-center align-items-center">
                    <div className="col-12 col-md-9 col-lg-7 col-xl-6">
                    <div className="card" style={{borderRadius: "15px"}}>
                        <div className="card-body px-5 py-2">
                        <h2 className="text-uppercase text-center mb-5 clipImage">Create an account</h2>
                        {isSuccessRegister && isSuccessRegister === "success" ?
                                <div className="alert alert-success mb-5">
                                <strong>Successfully you have registered!</strong>
                                </div>:isSuccessRegister && isSuccessRegister === "error" ?<div className="alert alert-danger">
                                <strong>Something went wrong!</strong>
                                </div>:null
                        }
                        <form onSubmit={formik.handleSubmit}>
                            <div className="form-outline input-group">
                                <span className="btn btn-outline-secondary form-field-mandatory-icon"><i className="bi bi-person-fill"></i></span>   
                                <input type="text" data-testid="name" id="name" name="name" onChange={formik.handleChange} value={formik.values.name} onBlur={formik.handleBlur} className="form-control form-control-md" placeholder="Your Name!" />
                            </div>
                            {formik.touched.name && formik.errors.name ?<label className="form-label text-danger"  data-testid="errorLabel" htmlFor="form3Example1cg">{formik.errors.name}</label>:null}
                            <div className="form-outline input-group mt-3">
                                <span className="btn btn-outline-secondary form-field-mandatory-icon"><i className="bi bi-envelope-at-fill"></i></span>   
                                <input type="email" data-testid="email" id="email" name="email" onChange={formik.handleChange} value={formik.values.email} onBlur={formik.handleBlur} className="form-control form-control-md" placeholder="Your Emailid!"/>
                            </div>
                            {formik.touched.email && formik.errors.password ? <label className="form-label text-danger"  data-testid="errorLabel" htmlFor="form3Example1cg">{formik.errors.password}</label>:null}
                            <div className="form-outline input-group mt-3">
                                <span className="btn btn-outline-secondary form-field-mandatory-icon"><i className="bi bi-shield-lock-fill"></i></span>    
                                <input type="password" data-testid="password" id="password" name="password" onChange={formik.handleChange} value={formik.values.password} onBlur={formik.handleBlur} className="form-control form-control-md" placeholder="Strong Password!" />
                            </div>
                            {formik.touched.password && formik.errors.password ? <label className="form-label text-danger"  data-testid="errorLabel" htmlFor="form3Example4cg">{formik.errors.password }</label>:null}
                            <div className="form-outline input-group mt-3">
                                <span className="btn btn-outline-secondary form-field-mandatory-icon"><i className="bi bi-key-fill"></i></span>    
                                <input type="password" data-testid="rePassword" id="rePassword" name="rePassword" onChange={formik.handleChange} value={formik.values.rePassword} onBlur={formik.handleBlur} className="form-control form-control-md" placeholder="Repeat Your Password"/>
                            </div>
                            {formik.touched.rePassword && formik.errors.telphone ? <label className="form-label text-danger"  data-testid="errorLabel" htmlFor="form3Example4cg">{formik.errors.telphone}</label>:null}
                            <div className="form-outline input-group mt-3">
                                <span className="btn btn-outline-secondary form-field-mandatory-icon"><i className="bi bi-telephone-fill"></i></span>    
                                <input type="tel" data-testid="telphone" id="telphone" name="telphone" onChange={formik.handleChange} value={formik.values.telphone} onBlur={formik.handleBlur} className="form-control form-control-md" placeholder="Your Mobile Number"/>
                            </div>
                            {formik.touched.telphone && formik.errors.telphone ? <label className="form-label text-danger"  data-testid="errorLabel" htmlFor="form3Example4cdg">{formik.errors.telphone}</label>:null}
                            <div className="form-outline input-group mt-3">
                                <span className="btn btn-outline-secondary form-field-mandatory-icon"><i className="bi bi-calendar-date-fill"></i></span>    
                                <input type="date" data-testid="birthdayDate" id="birthdayDate" name="birthdayDate" onChange={formik.handleChange} value={formik.values.birthdayDate} onBlur={formik.handleBlur}className="form-control form-control-md" placeholder="Your Date of Birth"/>
                            </div>
                            {formik.touched.birthdayDate && formik.errors.birthdayDate ?<label data-testid="errorLabel" className="form-label text-danger mb-4" htmlFor="form3Example4cdg">{ formik.errors.birthdayDate}</label>:null}
                            <div className="row form-outline input-group mt-3">
                            <span className="col-md-2 form-field-mandatory-icon">Gender:</span>
                            <div className="col-md-10">
                                <div className="form-check form-check-inline">
                                    <input 
                                    className="form-check-input"
                                    data-testid="femaleGender" 
                                    type="radio" 
                                    name="gender" 
                                    id="femaleGender"
                                    value="female"
                                    onChange={formik.handleChange} 
                                    />
                                    <label className="form-check-label" htmlFor="femaleGender">Female</label>
                                </div>

                                <div className="form-check form-check-inline">
                                    <input className="form-check-input" data-testid="maleGender"  type="radio" name="gender" id="maleGender" onChange={formik.handleChange} 
                                    value="male" />
                                    <label className="form-check-label" htmlFor="maleGender">Male</label>
                                </div>

                                <div className="form-check form-check-inline">
                                    <input className="form-check-input" data-testid="otherGender"  type="radio" name="gender" id="otherGender" onChange={formik.handleChange} 
                                    value="other" />
                                    <label className="form-check-label" htmlFor="otherGender">Other</label>
                                </div>
                            </div>
                            </div> 
                            {formik.touched.gender && formik.errors.gender ? <label  data-testid="errorLabel" className="form-label text-danger" htmlFor="form3Example4cdg">{formik.errors.gender}</label>:null}
                            <div className="form-check d-flex justify-content-center mb-4">
                            <input className="form-check-input me-2" data-testid="termOfService"  type="checkbox" value="" id="form2Example3cg" />
                            <label className="form-check-label" htmlFor="form2Example3g">
                                I agree all statements in <a href="#!" className="text-body"><u>Terms of service</u></a>
                            </label>
                            </div>

                            <div className="d-flex justify-content-center">
                            <input type="submit" data-testid="submitbtn" 
                                className="btn btn-success form-control form-control-md btn-block btn-lg gradient-custom-4 text-body" value={"Register"}></input>
                            </div>

                            <p className="text-center text-muted mt-5 mb-0">Have already an account? <a href="#!"
                                className="fw-bold text-body"><u>Login here</u></a></p>

                        </form>

                        </div>
                    </div>
                    </div>
                </div>
                </div>
            </div>
       </section>
       <Footer></Footer>
    </>)
}

export default Signup;