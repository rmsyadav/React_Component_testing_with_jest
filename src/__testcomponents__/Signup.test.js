import { rest } from 'msw';
import renderer from 'react-test-renderer';
import { render,screen,act } from '@testing-library/react';
import { setupServer } from 'msw/node';
import Signup from '../Components/Signup';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event'

const mockTasks ={statusCode:200,statusMessage:"success!"}
const mockTasks2 ={statusCode:403,statusMessage:"error!"}

const getTasksPath ='/shopping/api/v1/register';

const formHandler = rest.post(getTasksPath, async (req, res, ctx) =>{
  const  body = await req.json();
  const {email,password} = await body;
  if(email === "rmsyadav678@gmail.com")
  {
    return  res(ctx.json(mockTasks2)) 
  }
  else{
    return  res(ctx.json(mockTasks)) ;
  }  
 }  
);

const mswServer = setupServer(formHandler);



describe('SignUp component testing', () => {

   beforeAll(() => mswServer.listen());
   afterEach(() => mswServer.resetHandlers());
   afterAll(() => mswServer.close());
   
   it('SignUp component snapshot testing ',()=>{

     const signup = renderer.create(<BrowserRouter><Signup></Signup></BrowserRouter>).toJSON();
     expect(signup).toMatchSnapshot();
   })

   it('succesfully registered to my website!', async () => {
    const user = userEvent.setup();
     render(<BrowserRouter><Signup></Signup></BrowserRouter>);

     //**********************name******************
     const nameTextBoxElement = screen.getByTestId("name");
     await act(async()=>{
      await user.type(nameTextBoxElement,'Ramashankar kumar');
     }) 
     expect(screen.getByTestId('name')).toHaveValue('Ramashankar kumar');

     //******************email**************************
     const textBoxElement = screen.getByTestId("email");
     await act(async()=>{
      await user.type(textBoxElement,'rmsyadav567@gmail.com');
     }) 
     expect(screen.getByTestId('email')).toHaveValue('rmsyadav567@gmail.com');

     //*******************Password ****************
     const passwordElement = screen.getByTestId("password");
     await act(async()=>{
      await user.type(passwordElement,'rmsyadav@567');
     })
     expect(screen.getByTestId('password')).toHaveValue('rmsyadav@567');

     //*******************rePassword****************************

     const rePasswordElement = screen.getByTestId("rePassword");
     await act(async()=>{
      await user.type(rePasswordElement,'rmsyadav@567');
     })
     expect(rePasswordElement).toHaveValue('rmsyadav@567');

     //******************* Mobile number  **********************
     const phoneElement = screen.getByTestId("telphone");
     await act(async()=>{
      await user.type(phoneElement,'8839847229');
     })
     expect(phoneElement).toHaveValue('8839847229');
     

     //******************* date of birth  **********************
     const birthdayDateElement = screen.getByTestId("birthdayDate");
     //await user.clear()
     await act(async()=>{
      await user.type(birthdayDateElement,'2000-01-01');
     })
     expect(birthdayDateElement).toHaveValue('2000-01-01');
     
     //******************* select gender  **********************
     const genderElement = screen.getByTestId("maleGender");
     await act(async()=>{
      await user.click(genderElement);
     })
     expect(genderElement).toHaveFocus();
     expect(genderElement).toBeChecked();

     //******************* Submit button  **********************
     const submitBtnElement = screen.getByTestId('submitbtn');
     await act(async()=>{
      await user.click(submitBtnElement);
     }) 
     const displayedSuccessMessage = await screen.findByText("Successfully you have registered!");
     expect(displayedSuccessMessage).toBeInTheDocument();
   });

   it('Failed to registration to my website!', async () => {
    const user = userEvent.setup();
     render(<BrowserRouter><Signup></Signup></BrowserRouter>);

     //**********************name******************
     const nameTextBoxElement = screen.getByTestId("name");
     await act(async()=>{
      await user.type(nameTextBoxElement,'Ramashankar kumar');
     }) 
     expect(screen.getByTestId('name')).toHaveValue('Ramashankar kumar');

     //******************email**************************
     const textBoxElement = screen.getByTestId("email");
     await act(async()=>{
      await user.type(textBoxElement,'rmsyadav678@gmail.com');
     }) 
     expect(screen.getByTestId('email')).toHaveValue('rmsyadav678@gmail.com');

     //*******************Password ****************
     const passwordElement = screen.getByTestId("password");
     await act(async()=>{
      await user.type(passwordElement,'rmsyadav@567');
     })
     expect(screen.getByTestId('password')).toHaveValue('rmsyadav@567');

     //*******************rePassword****************************

     const rePasswordElement = screen.getByTestId("rePassword");
     await act(async()=>{
      await user.type(rePasswordElement,'rmsyadav@567');
     })
     expect(rePasswordElement).toHaveValue('rmsyadav@567');

     //******************* Mobile number  **********************
     const phoneElement = screen.getByTestId("telphone");
     await act(async()=>{
      await user.type(phoneElement,'8839847229');
     })
     expect(phoneElement).toHaveValue('8839847229');
     

     //******************* date of birth  **********************
     const birthdayDateElement = screen.getByTestId("birthdayDate");
     //await user.clear()
     await act(async()=>{
      await user.type(birthdayDateElement,'2000-01-01');
     })
     expect(birthdayDateElement).toHaveValue('2000-01-01');
     
     //******************* select gender  **********************
     const genderElement = screen.getByTestId("maleGender");
     await act(async()=>{
      await user.click(genderElement);
     })
     expect(genderElement).toHaveFocus();
     expect(genderElement).toBeChecked();

     //******************* Submit button  **********************
     const submitBtnElement = screen.getByTestId('submitbtn');
     await act(async()=>{
      await user.click(submitBtnElement);
     }) 
     const displayedSuccessMessage = await screen.findByText("Something went wrong!");
     expect(displayedSuccessMessage).toBeInTheDocument();
   });

   it('Failed to validation of registration form! when name field as empty=>', async () => {
    const user = userEvent.setup();
     render(<BrowserRouter><Signup></Signup></BrowserRouter>);     

     //******************email**************************
     const textBoxElement = screen.getByTestId("email");
     await act(async()=>{
      await user.type(textBoxElement,'rmsyadav678@gmail.com');
     }) 
     expect(screen.getByTestId('email')).toHaveValue('rmsyadav678@gmail.com');

     //*******************Password ****************
     const passwordElement = screen.getByTestId("password");
     await act(async()=>{
      await user.type(passwordElement,'rmsyadav@567');
     })
     expect(screen.getByTestId('password')).toHaveValue('rmsyadav@567');

     //*******************rePassword****************************

     const rePasswordElement = screen.getByTestId("rePassword");
     await act(async()=>{
      await user.type(rePasswordElement,'rmsyadav@567');
     })
     expect(rePasswordElement).toHaveValue('rmsyadav@567');

     //******************* Mobile number  **********************
     const phoneElement = screen.getByTestId("telphone");
     await act(async()=>{
      await user.type(phoneElement,'8839847229');
     })
     expect(phoneElement).toHaveValue('8839847229');
     

     //******************* date of birth  **********************
     const birthdayDateElement = screen.getByTestId("birthdayDate");
     //await user.clear()
     await act(async()=>{
      await user.type(birthdayDateElement,'2000-01-01');
     })
     expect(birthdayDateElement).toHaveValue('2000-01-01');
     
     //******************* select gender  **********************
     const genderElement = screen.getByTestId("maleGender");
     await act(async()=>{
      await user.click(genderElement);
     })
     expect(genderElement).toHaveFocus();
     expect(genderElement).toBeChecked();

     //******************* Submit button  **********************
     const submitBtnElement = screen.getByTestId('submitbtn');
     await act(async()=>{
      await user.click(submitBtnElement);
     }) 
     const displayedSuccessMessage = await screen.findByText("Required");
     expect(displayedSuccessMessage).toBeInTheDocument();
   });

   it('Failed to registration to my website!', async () => {
    const user = userEvent.setup();
     render(<BrowserRouter><Signup></Signup></BrowserRouter>);

     //**********************name******************
     const nameTextBoxElement = screen.getByTestId("name");
     await act(async()=>{
      await user.type(nameTextBoxElement,'Ramashankar kumar yadav ji hello ji');
     }) 
     expect(screen.getByTestId('name')).toHaveValue('Ramashankar kumar yadav ji hello ji');

     //******************email**************************
     const textBoxElement = screen.getByTestId("email");
     await act(async()=>{
      await user.type(textBoxElement,'rmsyadav678@gmail.com');
     }) 
     expect(screen.getByTestId('email')).toHaveValue('rmsyadav678@gmail.com');

     //*******************Password ****************
     const passwordElement = screen.getByTestId("password");
     await act(async()=>{
      await user.type(passwordElement,'rmsyadav@567');
     })
     expect(screen.getByTestId('password')).toHaveValue('rmsyadav@567');

     //*******************rePassword****************************

     const rePasswordElement = screen.getByTestId("rePassword");
     await act(async()=>{
      await user.type(rePasswordElement,'rmsyadav@567');
     })
     expect(rePasswordElement).toHaveValue('rmsyadav@567');

     //******************* Mobile number  **********************
     const phoneElement = screen.getByTestId("telphone");
     await act(async()=>{
      await user.type(phoneElement,'8839847229');
     })
     expect(phoneElement).toHaveValue('8839847229');
     

     //******************* date of birth  **********************
     const birthdayDateElement = screen.getByTestId("birthdayDate");
     //await user.clear()
     await act(async()=>{
      await user.type(birthdayDateElement,'2000-01-01');
     })
     expect(birthdayDateElement).toHaveValue('2000-01-01');
     
     //******************* select gender  **********************
     const genderElement = screen.getByTestId("maleGender");
     await act(async()=>{
      await user.click(genderElement);
     })
     expect(genderElement).toHaveFocus();
     expect(genderElement).toBeChecked();

     //******************* Submit button  **********************
     const submitBtnElement = screen.getByTestId('submitbtn');
     await act(async()=>{
      await user.click(submitBtnElement);
     }) 
     const displayedSuccessMessage = await screen.findByText("Must be 25 characters or less");
     expect(displayedSuccessMessage).toBeInTheDocument();
   });
   it('Failed to registration to my website!', async () => {
    const user = userEvent.setup();
     render(<BrowserRouter><Signup></Signup></BrowserRouter>);
     //******************* Submit button  **********************
     const submitBtnElement = screen.getByTestId('submitbtn');
     await act(async()=>{
      await user.click(submitBtnElement);
     }) 
     const displayedSuccessMessage = await screen.findAllByTestId("errorLabel");
     expect(displayedSuccessMessage).toHaveLength(7);
   });
 
 });