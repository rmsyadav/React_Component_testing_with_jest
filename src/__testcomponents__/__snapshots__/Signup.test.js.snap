// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component testing SignUp component snapshot testing  1`] = `
Array [
  <section
    className="container-fluid bg-image rounded-3"
    style={
      Object {
        "backgroundImage": "url('https://mdbcdn.b-cdn.net/img/Photos/new-templates/search-box/img4.webp')",
      }
    }
  >
    <div
      className="mask py-5 d-flex align-items-center container gradient-custom-3"
    >
      <div
        className="container"
      >
        <div
          className="row d-flex justify-content-center align-items-center"
        >
          <div
            className="col-12 col-md-9 col-lg-7 col-xl-6"
          >
            <div
              className="card"
              style={
                Object {
                  "borderRadius": "15px",
                }
              }
            >
              <div
                className="card-body px-5 py-2"
              >
                <h2
                  className="text-uppercase text-center mb-5 clipImage"
                >
                  Create an account
                </h2>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="form-outline input-group"
                  >
                    <span
                      className="btn btn-outline-secondary form-field-mandatory-icon"
                    >
                      <i
                        className="bi bi-person-fill"
                      />
                    </span>
                    <input
                      className="form-control form-control-md"
                      data-testid="name"
                      id="name"
                      name="name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Your Name!"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-outline input-group mt-3"
                  >
                    <span
                      className="btn btn-outline-secondary form-field-mandatory-icon"
                    >
                      <i
                        className="bi bi-envelope-at-fill"
                      />
                    </span>
                    <input
                      className="form-control form-control-md"
                      data-testid="email"
                      id="email"
                      name="email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Your Emailid!"
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    className="form-outline input-group mt-3"
                  >
                    <span
                      className="btn btn-outline-secondary form-field-mandatory-icon"
                    >
                      <i
                        className="bi bi-shield-lock-fill"
                      />
                    </span>
                    <input
                      className="form-control form-control-md"
                      data-testid="password"
                      id="password"
                      name="password"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Strong Password!"
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    className="form-outline input-group mt-3"
                  >
                    <span
                      className="btn btn-outline-secondary form-field-mandatory-icon"
                    >
                      <i
                        className="bi bi-key-fill"
                      />
                    </span>
                    <input
                      className="form-control form-control-md"
                      data-testid="rePassword"
                      id="rePassword"
                      name="rePassword"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Repeat Your Password"
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    className="form-outline input-group mt-3"
                  >
                    <span
                      className="btn btn-outline-secondary form-field-mandatory-icon"
                    >
                      <i
                        className="bi bi-telephone-fill"
                      />
                    </span>
                    <input
                      className="form-control form-control-md"
                      data-testid="telphone"
                      id="telphone"
                      name="telphone"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Your Mobile Number"
                      type="tel"
                      value=""
                    />
                  </div>
                  <div
                    className="form-outline input-group mt-3"
                  >
                    <span
                      className="btn btn-outline-secondary form-field-mandatory-icon"
                    >
                      <i
                        className="bi bi-calendar-date-fill"
                      />
                    </span>
                    <input
                      className="form-control form-control-md"
                      data-testid="birthdayDate"
                      id="birthdayDate"
                      name="birthdayDate"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Your Date of Birth"
                      type="date"
                      value=""
                    />
                  </div>
                  <div
                    className="row form-outline input-group mt-3"
                  >
                    <span
                      className="col-md-2 form-field-mandatory-icon"
                    >
                      Gender:
                    </span>
                    <div
                      className="col-md-10"
                    >
                      <div
                        className="form-check form-check-inline"
                      >
                        <input
                          className="form-check-input"
                          data-testid="femaleGender"
                          id="femaleGender"
                          name="gender"
                          onChange={[Function]}
                          type="radio"
                          value="female"
                        />
                        <label
                          className="form-check-label"
                          htmlFor="femaleGender"
                        >
                          Female
                        </label>
                      </div>
                      <div
                        className="form-check form-check-inline"
                      >
                        <input
                          className="form-check-input"
                          data-testid="maleGender"
                          id="maleGender"
                          name="gender"
                          onChange={[Function]}
                          type="radio"
                          value="male"
                        />
                        <label
                          className="form-check-label"
                          htmlFor="maleGender"
                        >
                          Male
                        </label>
                      </div>
                      <div
                        className="form-check form-check-inline"
                      >
                        <input
                          className="form-check-input"
                          data-testid="otherGender"
                          id="otherGender"
                          name="gender"
                          onChange={[Function]}
                          type="radio"
                          value="other"
                        />
                        <label
                          className="form-check-label"
                          htmlFor="otherGender"
                        >
                          Other
                        </label>
                      </div>
                    </div>
                  </div>
                  <div
                    className="form-check d-flex justify-content-center mb-4"
                  >
                    <input
                      className="form-check-input me-2"
                      data-testid="termOfService"
                      id="form2Example3cg"
                      type="checkbox"
                      value=""
                    />
                    <label
                      className="form-check-label"
                      htmlFor="form2Example3g"
                    >
                      I agree all statements in 
                      <a
                        className="text-body"
                        href="#!"
                      >
                        <u>
                          Terms of service
                        </u>
                      </a>
                    </label>
                  </div>
                  <div
                    className="d-flex justify-content-center"
                  >
                    <input
                      className="btn btn-success form-control form-control-md btn-block btn-lg gradient-custom-4 text-body"
                      data-testid="submitbtn"
                      type="submit"
                      value="Register"
                    />
                  </div>
                  <p
                    className="text-center text-muted mt-5 mb-0"
                  >
                    Have already an account? 
                    <a
                      className="fw-bold text-body"
                      href="#!"
                    >
                      <u>
                        Login here
                      </u>
                    </a>
                  </p>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <footer
    className="container-fluid footer-img rounded-2"
  >
    <div
      className="text-center footer-fontsize-mobile"
    >
      © 2023 Copyright:
      <span
        className="text-reset fw-bold"
      >
         Learning App
      </span>
    </div>
  </footer>,
]
`;
